### DISCUSSION TITLE
How to remove deploy key from deleted repo?

### DISCUSSION THREAD

**Discussion by faaik** [community/community#21993] [New to GitHub]
<p>I have a deploy key added to the repo that I have deleted, unfortunately, I did not delete the key prior to deleting the repo. Now I have a problem, where I can’t add the key to the new repo, since it is saying <code>Key is already in use</code>.</p>
<ul>
<li>I can’t regenerate the ssh key, since it’s provided by SaaS company;</li>
</ul>

**Comment by airtower-luna** [comment]
<p>That sound like a bug to me! :beetle:</p>
<p>You might want to <a href="https://support.github.com/contact/bug-report">file a bug report</a>, and I hope some of the staff members here see this.</p>

**Comment by prodkt** [comment]
<p>I have this same issue. Deleted Repo’s you cannot remove DEPLOY Keys. Problem is those deleted repo’s have been regenerated with the same name. So you cannot restore deleted repo’s to evaluate and remove deploy keys from those projects. :-1:</p>

**Comment by amityweb** [comment]
<p>Is there a fix to this yet? I created a repo, added a deployment key, realised I made some mistakes and wanted to start again so deleted the repo, and in my new repo I hit this issue right away! I cannot proceed, as the deployment key is auto generated and is the same as the one attached to the deleted repo.</p>

**Comment by faaik** [comment, marked as answer]
<p><a class="mention" href="/amityweb">@amityweb</a> and <a class="mention" href="/prodkt">@prodkt</a> GitHub repos are “soft” deleted, so the best way is to create support ticket and ask for purging the repo. That’s what I did at the end.</p>

**Reply by BoyetDgte** [reply]
How did you contact them? @thelittlehawk Thanks!

**Reply by airtower-luna** [reply]
The support form is here: https://support.github.com/request

**Comment by bartclaeys** [comment]
Had the same problem, luckily I could restore the deleted repo through Github Settings > Repositories (it was less than 90 days ago) and remove the deployment key.

**Reply by ZanyRain** [reply]
> Had the same problem, luckily I could restore the deleted repo through Github Settings > Repositories (it was less than 90 days ago) and remove the deployment key.

Maybe it's a github feature :)
I meet this problem too.

**Comment by BoyetDgte** [comment]
(+1) plus1 to this BUG. But for me, I have to wait 1 hour before the deleted repos are showing in the deleted repositories tab...

**Comment by appsscriptdev** [comment]
+1

**Comment by rdwrome** [comment]
+1
